tests/v2/validation/pipeline/releaseupgrade/main.go

```diff
@@ -3,73 +3,98 @@ package main
+const (
+	dirName = "cattle-configs"
 
-func main() {
-	releaseUpgradeConfig := new(pipelineutil.ReleaseUpgradeConfig)
+	nodeProviderFileName = "node"
+	customFileName       = "custom"
+
+	rke1FileName = "rke1"
+	rke2FileName = "rke2"
+	k3sFileName  = "k3s"
+
+	eksFileName = "eks"
+	gkeFileName = "gke"
+	aksFileName = "aks"
+)
 
+func main() {
 	if configPath == "" || adminToken == "" {
-		logrus.Fatalf("error no file or token in config path")
+		logrus.Fatalf("error file in config path or token doesn't exist")
 	}
 
-	parseYaml(configPath, releaseUpgradeConfig)
+	os.Setenv(configEnvironmentKey, configPath)
+	defer os.Unsetenv(configEnvironmentKey)
 
-	releaseUpgradeConfig.MainConfig.RancherConfig.Host = releaseUpgradeConfig.HAConfig.Host
-	releaseUpgradeConfig.MainConfig.RancherConfig.AdminToken = adminToken
+	haConfig := new(pipelineutil.HAConfig)
+	config.LoadConfig(pipelineutil.HAConfigKey, haConfig)
 
-	//Overwrite/update flag to grab cluster names that are provisioned
-	releaseUpgradeConfig.MainConfig.EnvironmentFlags.DesiredFlags = environmentflag.UpdateClusterName.String()
+	rancherConfig := new(rancher.Config)
+	config.LoadConfig(rancher.ConfigurationFileKey, rancherConfig)
+
+	rancherConfig.Host = haConfig.Host
+	rancherConfig.AdminToken = adminToken
 
 	//Rancher cleanup has to be false for the future steps to prevent resource deletion
 	rancherCleanup := false
-	releaseUpgradeConfig.MainConfig.RancherConfig.Cleanup = &rancherCleanup
+	rancherConfig.Cleanup = &rancherCleanup
 
 	//HA cleanup default to true if not specified
-	if releaseUpgradeConfig.HAConfig.Cleanup == nil {
+	if haConfig.Cleanup == nil {
 		HACleanup := true
-		releaseUpgradeConfig.HAConfig.Cleanup = &HACleanup
+		haConfig.Cleanup = &HACleanup
 	}
 
 	//Rancher insecure default to true if not specified
-	if releaseUpgradeConfig.HAConfig.Insecure == nil {
+	if haConfig.Insecure == nil {
 		insecure := true
-		releaseUpgradeConfig.MainConfig.RancherConfig.Insecure = &insecure
-		releaseUpgradeConfig.HAConfig.Insecure = &insecure
+		rancherConfig.Insecure = &insecure
+		haConfig.Insecure = &insecure
 	} else {
-		releaseUpgradeConfig.MainConfig.RancherConfig.Insecure = releaseUpgradeConfig.HAConfig.Insecure
+		rancherConfig.Insecure = haConfig.Insecure
 	}
 
-	//Upgrade package and tests
-	releaseUpgradeConfig.UpgradeTestPackage = "upgrade"
-	releaseUpgradeConfig.PreUpgradeTestCase = `-run \"TestWorkloadUpgradeTestSuite/TestWorkloadPreUpgrade\"`
-	releaseUpgradeConfig.PostUpgradeTestCase = `-run \"TestWorkloadUpgradeTestSuite/TestWorkloadPostUpgrade\"`
-	releaseUpgradeConfig.UpgradeKubernetesTestCase = `-run \"TestKubernetesUpgradeTestSuite/TestUpgradeKubernetes\"`
+	config.UpdateConfig(rancher.ConfigurationFileKey, rancherConfig)
+	config.UpdateConfig(pipelineutil.HAConfigKey, haConfig)
+
+	testCases := new(pipelineutil.TestCases)
+	config.LoadAndUpdateConfig(pipelineutil.TestCasesConfigKey, testCases, func() {
+		//TODO update Jenkinsfile to nested struct
```

Comment:
Forgot this TODO item, it's updated but the comment needs to be removed with the upcoming squash.

Reply:
Removed